JavaScript/JavaScript.sublime-syntax

```diff
@@ -106,6 +106,8 @@ variables:
 
   # '@' followed by a pattern like \S but excluding literal '*' and '@'.
   jsdoc_block_tag: \@[^\n\t\f\v *@]+
+  # Tags that are optionally followed by a type expression.
+  jsdoc_block_tag_with_type: \@(?:arg(?:ument|)|param|type(?:def|)|returns?|prop(?:erty|)|enum|implements|member|module|namespace|(?:pr(?:ivate|otected))|template|this|throws|exception|yields?|var)\b
```

Comment:
Generally literals are very easy for regex engines to optimize themselves, so we tend to make them more readable and searchable in this repository :)
Also a small style thing, empty alternations can be confusing (i.e. maybe something is missing) but with optional quantifiers the intentions are clear
```suggestion
  jsdoc_block_tag_with_type: \@(?:arg(?:ument)?|param|typedef|type|returns?|prop(?:erty)?|enum|implements|member|module|namespace|private|protected|template|this|throws|exception|yields?|var)\b
```